cassandra-sink/src/main/java/org/springframework/xd/sink/cassandra/CassandraSinkOptionsMetaData.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.sink.cassandra;
+
+import javax.validation.constraints.AssertFalse;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.cassandra.core.ConsistencyLevel;
+import org.springframework.cassandra.core.RetryPolicy;
+import org.springframework.integration.cassandra.outbound.CassandraMessageHandler;
+import org.springframework.util.StringUtils;
+import org.springframework.xd.module.options.mixin.CassandraConnectionMixin;
+import org.springframework.xd.module.options.spi.Mixin;
+import org.springframework.xd.module.options.spi.ModuleOption;
+
+/**
+ * @author Artem Bilan
+ */
+@Mixin(CassandraConnectionMixin.class)
+public class CassandraSinkOptionsMetaData {
+
+	private ConsistencyLevel consistencyLevel;
+
+	private RetryPolicy retryPolicy;
+
+	private Integer ttl;
+
+	private CassandraMessageHandler.Type queryType = CassandraMessageHandler.Type.INSERT;
+
+	private String ingestQuery;
+
+	private String statementExpression;
+
+	@ModuleOption("the consistencyLevel option of WriteOptions")
+	public void setConsistencyLevel(ConsistencyLevel consistencyLevel) {
+		this.consistencyLevel = consistencyLevel;
+	}
+
+	@ModuleOption("the retryPolicy  option of WriteOptions")
+	public void setRetryPolicy(RetryPolicy retryPolicy) {
+		this.retryPolicy = retryPolicy;
+	}
+
+	@ModuleOption("the time-to-live option of WriteOptions")
+	public void setTtl(Integer ttl) {
+		this.ttl = ttl;
+	}
+
+	@ModuleOption("the queryType for Cassandra Sink. Defaults to INSERT")
```

Comment:
Don't mention the default in description, this is looked up automatically (and could be overridden)